main/utils.zkasm

```diff
@@ -990,122 +990,306 @@ invalidCall:
     0                               :MSTORE(SP++), JMP(readCode)
 
 VAR GLOBAL pushBytes
-VAR GLOBAL numBlocks
-VAR GLOBAL leftBytes
-VAR GLOBAL accumulator
+VAR GLOBAL pushBytes_RR
 ; @info Read bytes opcode PUSHX
 ; @internalParam {numBlocks} 4 bytes blocks to read
 ; @internalParam {leftBytes} remaining bytes
-; @in D => bytes to read
+; @in D,E => bytes to read (1..32)
```

Comment:
Maybe I miss something but I don't see why is need D as input

Reply:
If we don't use D, means need to restore D value when function finish, only for this reason. For example, in case of push32, we don't save old value of D because we restore directly 32 over D. If we don't use D, need and extra step to store D before overwrite it.